v0.0.5 (#397)
apollographql/apollo-cli: CHANGELOG.md

## 🛠 Maintenance
## 📚 Documentation -->

# [0.0.5] - 2021-03-30

> > Important: X breaking changes below, indicated by **❗ BREAKING ❗**

Reviewer: ```suggestion
> Important: 2 breaking change below, indicated by **❗ BREAKING ❗**
```

## 🚀 Features
- **Adds introspection ability for subgraphs - [lrlna], [issue/349] [pull/377]**

  A new command, `rover subgraph introspect` has been added. This command
  runs a _federated introspection_ query against a server which has
  implemented the requirements of the [federation
  specification](https://www.apollographql.com/docs/federation/federation-spec/).

>   This command accepts endpoint [headers] for making the introspection

Reviewer: Missing the url for this

>   This command accepts endpoint [headers] for making the introspection

Author: ```suggestion
  This command accepts endpoint headers (`-H`, `--header`) for making the introspection
```

  request (if required) and outputs SDL to stdout.

  [lrlna]: https://github.com/lrlna

>   [pull/377]: https://github.com/apollographql/rover/pull/277

Reviewer: ```suggestion
  [pull/377]: https://github.com/apollographql/rover/pull/377
```

  [issue/349]: https://github.com/apollographql/rover/issues/349

> - ** Fallback to monochromic output when `tput` is unavailable - [abernix], [issue/371] [pull/372]**

Reviewer 2: ```suggestion
- **Fallback to monochromic output in installer when `tput` is unavailable - [abernix], [issue/371] [pull/372]**
```

Reviewer 2: not sure if we need the "in installer" but that space after the `**` will make the heading not bold

  The `tput` command allows easier ANSI output using named values in rather than
  control characters.

  While we could just use control characters and maintain colored output in the
  absence of `tput`, it's probably also reasonable to gracefully fall back to
  monochromatic output.

  [abernix]: https://github.com/abernix
  [pull/372]: https://github.com/apollographql/rover/pull/372
  [issue/371]: https://github.com/apollographql/rover/issues/371

## ❗ BREAKING ❗
- ** Renames `core build` to `supergraph compose` - [lrlna], [pull/391]**

>   To align with other Apollo teams on the usage `supergraph` and

Reviewer 2: ```suggestion
  To align with other Apollo teams on the usage of `supergraph` and
```

  `composition`, we are renaming `core build` to `supergraph compose`.

  [lrlna]: https://github.com/lrlna
  [pull/391]: https://github.com/apollographql/rover/pull/391

- ** Updates harmonizer@0.2.2 - [abernix], [pull/396]**

  Updates harmonizer to the latest version. This version now composes and
  returns a core schema instead of CSDL. CSDL was an internal implementation

>   of composition and this new format is meant to bring stability to `rover

Reviewer 2: ```suggestion
  of composition and this new format is meant to bring some stability to `rover
```
idk if we need to hedge too much here but i don't necessarily wanna promise _too_ much stability

>   supergraph compose.

Reviewer: ```suggestion
  supergraph compose`.
```

  [abernix]: https://github.com/abernix
  [pull/396]: https://github.com/apollographql/rover/pull/396

## 🐛 Fixes

> - ** handle 400-599 HTTP responses - [lrlna], [issue/394] [issue/187] [pull/395]**

Reviewer 2: ```suggestion
- **Handle 400-599 HTTP responses - [lrlna], [issue/394] [issue/187] [pull/395]**
```

>   We are not currently handling any unsuccessful requests wehn handling a
>   response from an endpoint. This fix makes sure we check for errors to
>   return those errors before trying to parse the response body.

Reviewer 2: ```suggestion
  Previously, Rover did not provide errors for any HTTP requests that return a status code between 400 and 599. This fix makes sure Rover checks for those errors before moving on to parsing the response body.
```

Reviewer 2: I _personally_ think this belongs squarely in the feature section! Better error messages feel like features to me. Not a hill I'll die on though :)

>   This fix also does an extra check for 400 error, as the Apollo

Reviewer 2: ```suggestion
  This fix also does an extra check for 400 errors, as the Apollo
```

>   Server sends additional information that we can display to the users.

Reviewer 2: ```suggestion
  Server sends additional information that we can display to users.
```

  [lrlna]: https://github.com/lrlna
  [issue/394]: https://github.com/apollographql/rover/issues/394
  [issue/187]: https://github.com/apollographql/rover/issues/187
  [pull/395]: https://github.com/apollographql/rover/pull/395

## 🛠 Maintenance

> - ** Sign and notarise MacOS binaries as part of CI - [EverlastingBugstopper], [pull/363]**

Reviewer 2: ```suggestion
- **Sign and notarize MacOS binaries as part of CI - [EverlastingBugstopper], [pull/363]**
```

>   This automates our signing and notarisation process when releasing MacOS

Reviewer 2: ```suggestion
  This automates our signing and notarization process when releasing MacOS
```

Author: 😬 My spellchecker is clearly getting me into trouble with people across the pond 😬

>   binaries. This is especially necessary to install and run rover on latest

Reviewer 2: ```suggestion
  binaries. This is especially necessary to install and run Rover on latest
```

  M1s and Big Sur.

  [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
  [pull/363]: https://github.com/apollographql/rover/pull/363

> - ** test, build and release on ubuntu-16.04 - [abernix], [pull/381]**

Reviewer 2: ```suggestion
- **Test, build and release on ubuntu-16.04 - [abernix], [pull/381]**
```

  This pins us to Ubuntu 16.04 which ships with glib 2.19. This should allow
  us to work with a wider range of operating systems than the newer glib
  that we get with Ubuntu 20.04, which is ubuntu-latest on GitHub Actions
  Virtual Environments (which resulted in a Rover that wouldn't run on
  Ubuntu 18.04).

  Ubuntu 16.04 is LTS until April 2024, and is still receiving active
  updates through the LTS program.

  [abernix]: https://github.com/abernix
  [pull/381]: https://github.com/apollographql/rover/pull/381

> - ** Adds rust caching to clippy - [EverlastingBugstopper], [pull/365]**

Reviewer 2: ```suggestion
- **Cache Rust artifacts in CI linter job- [EverlastingBugstopper], [pull/365]**
```

  The rest of our GitHub actions workflows pull from the cache to take

>   advantage of rust's incremental compilation. We now do this for clippy

Reviewer 2: ```suggestion
  advantage of Rust's incremental compilation. We now do this for clippy
```

  too so it finishes (and fails) faster.

  [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
  [pull/365]: https://github.com/apollographql/rover/pull/365

> - ** Addresses new clippy 1.51 warning - [EverlastingBugstopper], [pull/364]**

Reviewer 2: ```suggestion
- **Addresses new clippy 1.51 warning - [EverlastingBugstopper], [pull/364]**
```

>   New clippy update enforces rusty naming conventions and disallows
>   capitalizing acronyms in enum variants/struct names. We had to ignore those
>   clippy warnings in our rover-client/query module because all of it is
>   autogenerated and there's nothing we can do.

Reviewer 2: ```suggestion
  Addresses some stylistic problems noticed by the new version of our linter [clippy](https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-151)
```

>   We now also implement `From` instead of `Into` for GitContext since `From`
>   gives you `Into` for free while that is not the case the other way around.
> 

Reviewer 2: ```suggestion
```

don't think we need this tbh

  [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
  [pull/364]: https://github.com/apollographql/rover/pull/364

## 📚 Documentation

> - ** Update documentation for 0.0.5 release- [JakeDawkins] [StephenBarlow] [EverlastingBugstopper], [pull/389]**

Reviewer 2: ```suggestion
- **Update documentation for 0.0.5 release- [JakeDawkins] [StephenBarlow] [EverlastingBugstopper], [pull/389]**
```

>   Documents recent additions to rover in detail, including `rover supergraph

Reviewer 2: ```suggestion
  Documents recent additions to Rover in detail, including `rover supergraph
```
